### PR TITLE
[LLVM] bump to 12.0.1-4

### PR DESCRIPTION
**Pull request by vchuravy**
Includes https://github.com/JuliaLang/julia/pull/41439#issuecomment-885528292 and https://github.com/JuliaLang/julia/pull/42440#issuecomment-931905046

### CODE REVIEW

**Comment by DilumAluthge** [comment]
@staticfloat Looks like we are getting some Buildkite failures on amdci7 but not on amdci5. An example:

```
file                                (7) |         failed at 2021-10-06T00:49:38.236
Error During Test at /cache/build/amdci7-2/julialang/julia-master/julia-19134592a9/share/julia/test/testdefs.jl:21
  Got exception outside of a @test
  LoadError: IOError: chown("/tmp/jl_FQMcP7/bfile.txt", -2, -1): invalid argument (EINVAL)
  Stacktrace:
    [1] uv_error
      @ ./libuv.jl:97 [inlined]
    [2] chown(path::String, owner::Int64, group::Int64)
      @ Base.Filesystem ./file.jl:1168
    [3] top-level scope
      @ /cache/build/amdci7-2/julialang/julia-master/julia-19134592a9/share/julia/test/file.jl:535
    [4] include
      @ ./Base.jl:416 [inlined]
    [5] macro expansion
      @ /cache/build/amdci7-2/julialang/julia-master/julia-19134592a9/share/julia/test/testdefs.jl:24 [inlined]
    [6] macro expansion
      @ /cache/build/amdci7-5/julialang/julia-master/usr/share/julia/stdlib/v1.8/Test/src/Test.jl:1321 [inlined]
    [7] macro expansion
      @ /cache/build/amdci7-2/julialang/julia-master/julia-19134592a9/share/julia/test/testdefs.jl:23 [inlined]
    [8] macro expansion
      @ ./timing.jl:380 [inlined]
    [9] runtests(name::String, path::String, isolate::Bool; seed::UInt128)
      @ Main /cache/build/amdci7-2/julialang/julia-master/julia-19134592a9/share/julia/test/testdefs.jl:21
   [10] invokelatest(::Any, ::Any, ::Vararg{Any}; kwargs::Base.Pairs{Symbol, UInt128, Tuple{Symbol}, NamedTuple{(:seed,), Tuple{UInt128}}})
      @ Base ./essentials.jl:720
   [11] (::Distributed.var"#106#108"{Distributed.CallMsg{:call_fetch}})()
      @ Distributed /cache/build/amdci7-5/julialang/julia-master/usr/share/julia/stdlib/v1.8/Distributed/src/process_messages.jl:278
   [12] run_work_thunk(thunk::Distributed.var"#106#108"{Distributed.CallMsg{:call_fetch}}, print_error::Bool)
      @ Distributed /cache/build/amdci7-5/julialang/julia-master/usr/share/julia/stdlib/v1.8/Distributed/src/process_messages.jl:63
   [13] macro expansion
      @ /cache/build/amdci7-5/julialang/julia-master/usr/share/julia/stdlib/v1.8/Distributed/src/process_messages.jl:278 [inlined]
   [14] (::Distributed.var"#105#107"{Distributed.CallMsg{:call_fetch}, Distributed.MsgHeader, Sockets.TCPSocket})()
      @ Distributed ./task.jl:411
  in expression starting at /cache/build/amdci7-2/julialang/julia-master/julia-19134592a9/share/julia/test/file.jl:532
```

From https://buildkite.com/julialang/julia-master/builds/4398#4f9c9e09-f7da-4e0e-9f66-f8f4fa046843, which ran on amdci7.2.

The jobs that run on amdci5 don't seem to have the same error.

**Comment by DilumAluthge** [comment]
This looks similar to the problem we were trying to fix in https://github.com/JuliaLang/julia/pull/41682.

I'm not sure why it would be agent-dependent.

Also cc: @Keno since you helped out with this previously.

**Comment by DilumAluthge** [comment]
Seeing the same thing on amdci8, e.g. https://buildkite.com/julialang/julia-master/builds/4399#c580a08b-d431-4b17-adbb-1cea1a61a79a

**Comment by DilumAluthge** [comment]
For some reason, on amdci7 and amdci8, Julia thinks it is running as root inside the inner sandbox?

File: contrib/asan/Make.user.asan [hunk 1]
```diff
@@ -4,7 +4,7 @@ TOOLDIR=$(TOOLCHAIN)/usr/tools
 
 # use our new toolchain
 USECLANG=1
-override CC=$(BINDIR)/clang
+override CC=$(TOOLDIR)/clang
 override CXX=$(TOOLDIR)/clang++
```
[the remark this answers is not in the document]
Comment (vchuravy) [reply contrib/asan/Make.user.asan:8]: For asan it seems like it, but for tsan not quite yet

**Pull request merged**